charon/pkgs/maven.py

```diff
@@ -782,7 +783,7 @@ def _merge_directories_with_rename(src_dir: str, dest_dir: str, root: str):
                 _handle_archetype_catalog_merge(src_file, dest_file)
                 merged_count += 1
                 logger.debug("Merged archetype catalog: %s -> %s", src_file, dest_file)
-            if os.path.exists(dest_file):
+            elif os.path.exists(dest_file):
```

Comment:
@yma955 I'm not sure if this change will break your logic, but from my POV this should be elif but not single if. Please help confirm.